File: how-to-use-kotlin-in-java-project/src/main/java/dev/trile/customexception/SalaryCalculator.java
package dev.trile.customexception;

import static dev.trile.customexception.ErrorType.CALC_SALARY_PARAMS_INVALID;

public class SalaryCalculator {

  public int calcSalary(int workingDay, int salaryPerDay) throws SalaryException {

    if (workingDay <= 0)
      throw new SalaryException(CALC_SALARY_PARAMS_INVALID, "WorkingDay less than or equal zero");

    if (salaryPerDay <= 0)
      throw new SalaryException(CALC_SALARY_PARAMS_INVALID, "SalaryPerDay less than or equal zero");

    return workingDay * salaryPerDay;
  }
}

File: ioc-di/src/main/kotlin/dev/trile/iocdi/SalaryCalculator.kt
package dev.trile.iocdi

import org.springframework.stereotype.Component

@Component
class SalaryCalculator {

  fun calcSalary(workingDay: Int, salaryPerDay: Int): Int {
    if (workingDay <= 0) throw SalaryException(ErrorType.CALC_SALARY_PARAMS_INVALID, "WorkingDay less than or equal zero")
    if (salaryPerDay <= 0) throw SalaryException(ErrorType.CALC_SALARY_PARAMS_INVALID, "SalaryPerDay less than or equal zero")
    return workingDay * salaryPerDay
  }
}
